here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/SourceIdHandler.java

```diff
@@ -148,8 +141,8 @@ private static boolean sourceIdTransformationCapable(POp propertyOperation) {
         && propertyOperation.children() == null;
   }
 
-  private static boolean operationTypeAllowed(POp propertyOperation) {
-    return propertyOperation.op().equals(POpType.EQ)
-        || propertyOperation.op().equals(POpType.CONTAINS);
+  private static boolean operationTypeAllowed(PQuery propertyOperation) {
+    final AnyOp op = propertyOperation.getOp();
+    return op.equals(StringOp.EQUALS) || op.equals(StringOp.CONTAINS) || op.equals(DoubleOp.EQ);
```

Comment:
Maybe it's not harmful in this particular case, but in general when you compare `smePotentiallyNullObject` and `"CONSTANT"` it's better to invert the operation and start from constant. This way you'll be always safe from NPE.
In this case I would go with `StringOp.EQUALS.equals(op)` etc